From 0c25d57109ae7f7606d0e4c95f786ffd1fc7e3dc Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 22 Mar 2020 23:41:15 +0200
Subject: [PATCH] [SW-0] ppboot: add dev specific log of pcie wait

Add ability to print a log of the PCIe link waiting fact.
It is done in device specific manner, so each device can obey its
restrictions.
For example Goya is very sensitive to the PCIe link enablement timing,
so keep it clear - no log.
Others, add currently a minimalistic log of 3 characters: 5, LF, CR.

Change-Id: I5f39a9f16264ec331e1f4deafc3eac9e27bbb8d3
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/ppboot.c b/app/ppboot.c
index c1b7e1d..92137f4 100644
--- a/app/ppboot.c
+++ b/app/ppboot.c
@@ -103,6 +103,8 @@
 	return false;
 }
 
+__attribute__((weak)) void ppboot_log_waiting_for_pcie(void) {}
+
 static void ppboot_pcie_init(void)
 {
 	if (hl_pcie_link_up()) {
@@ -115,7 +117,8 @@
 		return;
 	}
 
-	//preboot_puts("Waiting for PCIe link...\n");
+	/* let the log be device specific according to its restrictions */
+	ppboot_log_waiting_for_pcie();
 	preboot_pcie_wait_for_link();
 	preboot_puts("PCIe link is UP!\n");
 }
diff --git a/gaudi/common.c b/gaudi/common.c
index 62ddc6f..5cb9397 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -202,3 +202,10 @@
 }
 
 void hl_plat_init(void) {}
+
+
+void ppboot_log_waiting_for_pcie(void)
+{
+	/* we are waiting for link, so do not waste too much time on log */
+	preboot_puts("5\n");
+}
diff --git a/goya2/common.c b/goya2/common.c
index 98c26a1..ad87908 100644
--- a/goya2/common.c
+++ b/goya2/common.c
@@ -205,6 +205,12 @@
 
 void hl_plat_init(void) {}
 
+void ppboot_log_waiting_for_pcie(void)
+{
+	/* we are waiting for link, so do not waste too much time on log */
+	preboot_puts("5\n");
+}
+
 void spi_ssi_cs0_get(unsigned int *mask, unsigned int *data_offset,
 		     unsigned int *out_en_offset)
 {
